internal/fileutils/fileutils.go

```diff
@@ -0,0 +1,41 @@
+// Package fileutils provides utility functions for file operations.
+package fileutils
+
+import (
+	"errors"
+	"io"
+	"os"
+)
+
+// FileExists checks if a file exists at the given path.
+func FileExists(path string) (bool, error) {
+	_, err := os.Stat(path)
+	if err != nil && !errors.Is(err, os.ErrNotExist) {
+		return false, err
+	}
+	return !errors.Is(err, os.ErrNotExist), nil
+}
+
+// IsDirEmpty checks if the specified directory is empty.
+func IsDirEmpty(path string) (bool, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return false, err
+	}
+	defer f.Close()
+
+	_, err = f.Readdirnames(1)
+	if errors.Is(err, io.EOF) {
+		return true, nil
+	}
+	return false, err
```

Comment:
`os.ReadDir` would create a slice with all the files in the directory, which would be inefficient since we only want to know if there are any files at all.

Reply:
Fair enough. I'm not a fan of choosing efficiency over readability, but the difference is minor for both cases so let's go with this.